google-ads-examples/src/main/java/com/google/ads/googleads/examples/planning/ForecastReach.java

```diff
@@ -0,0 +1,357 @@
+import com.google.ads.googleads.v4.services.ProductAllocation;
+import com.google.ads.googleads.v4.services.ProductMetadata;
+import com.google.ads.googleads.v4.services.ReachForecast;
+import com.google.ads.googleads.v4.services.ReachPlanServiceClient;
+import com.google.ads.googleads.v4.services.Targeting;
+import com.google.protobuf.BoolValue;
+import com.google.protobuf.Int32Value;
+import com.google.protobuf.Int64Value;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Demonstrates how to interact with the ReachPlanService to find plannable locations and product
+ * codes, build a media plan, and generate a video ads reach forecast.
+ */
+public class ForecastReach {
+
+  private static class ForecastReachParams extends CodeSampleParams {
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+  }
+
+  public static void main(String[] args) {
+    ForecastReachParams params = new ForecastReachParams();
+
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new ForecastReach().runExample(googleAdsClient, params.customerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID for the reach forecast.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId) {
+    String locationId = "2840"; // US
+    String currencyCode = "USD";
+    long budgetMicros = 500_000_000_000L;
```

Comment:
This is an awfully large number ($500,000). Was that intentional?

Reply:
Defer to @AnashOommen who wrote the golden.

Reply:
it *is* awfully large. Make it $5 pls...